apps/node/server.js

```diff
@@ -0,0 +1,138 @@
+const express = require("express");
+const axios = require("axios");
+const dotenv = require("dotenv").config();
+const crypto = require("crypto");
+const qs = require("qs");
+
+const ATTENTIVE_API_URL = "https://api.attentivemobile.com/v1";
+const ACCESS_TOKEN_ENDPOINT =
+  "https://api.attentivemobile.com/v1/authorization-codes/tokens";
+
+const CLIENT_ID = dotenv.parsed.CLIENT_ID; // Your client id
+const CLIENT_SECRET = dotenv.parsed.CLIENT_SECRET; // Your secret
+const REDIRECT_URI = dotenv.parsed.REDIRECT_URI; // Your redirect uri
+const PORT = 3002;
+
+let data = {};
+
+let app = express();
+app.use(express.json());
+
+app.get("/", (req, res) => {
+  res.send({ message: "Your Application is running successfully!" });
+});
+
+app.get("/install", async (req, res) => {
+  // Generates a random string containing numbers and letters
+  const state = generateRandomString(10);
+
+  // List of scopes that your app needs
+  const scopes = ["ecommerce:write", "events:write", "subscriptions:write"];
+
+  data[state] = {};
+  const redirect_url =
+    `https://ui.attentivemobile.com/integrations/oauth-install?client_id=${CLIENT_ID}` +
+    `&redirect_uri=${REDIRECT_URI}` +
+    `&scope=${scopes.join("+")}` +
+    `&state=${state}`;
+
+  res.redirect(redirect_url);
+});
+
+app.get("/callback", async (req, res) => {
```

Comment:
Express doesn't natively support async/await, these callbacks will swallow uncaught errors. This is a bad practice. I'd recommend either
1. Wrapping the entire payloads in a try/catch
2. Using another server library (IIRC this is essentially the `fastify` interface)
3. Moving the helper logic into a "service" class with async methods, making these callbacks into a thin adapter layer.

Reply:
I would just wrap the whole request handler in a `try/catch`

Reply:
done!